BossMod/Modules/RealmReborn/Trial/T09WhorleaterH/T09WhorleaterGenericActions.cs

```diff
@@ -0,0 +1,32 @@
+using BossMod.Components;
+
+namespace BossMod.Modules.RealmReborn.Trial.T09WhorleaterH;
+ 
+     class GrandFall : LocationTargetedAOEs
+    {
+        public GrandFall() : base(ActionID.MakeSpell(AID.GrandFall), 8) { }
+    }
+
+    class Hydroshot : PersistentVoidzoneAtCastTarget
+    {
+        public Hydroshot() : base(5, ActionID.MakeSpell(AID.Hydroshot), m => m.Enemies(OID.HydroshotZone), 0) { }
+    }
+    class Dreadstorm : PersistentVoidzoneAtCastTarget
+    {
+        public Dreadstorm() : base(5, ActionID.MakeSpell(AID.Dreadstorm), m => m.Enemies(OID.DreadstormZone), 0) { }
+    }
+    class T09WhorleaterHStates : StateMachineBuilder
```

Comment:
Typically I keep 'simple' components in the main .cs, next to the module, and don't move them out to 'generic actions'. Only if they tend to become big, I then move them to own .cs named after component (if there are several closely related ones - select most important one, and move it + related smaller ones together).

For states - similar, keep in main .cs if trivial, move out to xxxStates.cs if it becomes complex.